added suport for single line comment only languages and small QOL changes

alexm622 opened the pull request:
changes:
* added support for single line comment languages through the use of the
  constant `SINGLE_LINE_COMMENTS` in place of the postfix and uses the prefix as the comment
* added .gitignore to remove any binaries and temporary testing files from the git path
* added comments to `utils/comments.go` describing what type of file each of the lines corresponds to
* updated SupportedFileExtensions in `utils/constants.go` to better follow the standards of "supported file extensions", but i can't seem to find a place where this is implemented
## current issues
| issue | proposed solution |
|-------|----------------|
|Makefiles still do not work due to the program matching based off of extension and Makefiles not having an extension| a fix for this would be to rewrite the matching method to include full filenames, this would also have the side effect of being able to specify specific files in the config, which would then require a little more rewriting on matching the config file to the comment prefix/postfix in `comments.go`|
license header removal does not work for single line comments | needs a rework for files which support single line comments, maybe use start and end delimiters |

Marvin9 commented:
Good use cases which this project doesn't cover. 

I think for the standalone files (like Makefile), I would agree with the approach you suggested. This would be a perfect time to separate a function which performs adding/removing license given a file input. I am just wondering what are the config files other than `Makefile` does not have an extension.

About the second concern, I haven't thought much of that case. Depends on how we implement the addition, we just have to reverse engineer it.

That was brief perspective. I could not conclude without try and testing.

alexm622 committed:
support for single line comments fully implemented

single line comment support is properly implemented through the use of a DELIMITER constant

alexm622 commented:
working on better implementation
